s3fs/tests/test_s3fs.py

```diff
@@ -267,6 +267,17 @@ def test_move(s3):
     assert not s3.exists(fn)
 
 
+def test_get_put(s3):
+    import tempfile
+    fn = tempfile.mktemp()
+    s3.get(test_bucket_name+'/test/accounts.1.json', fn)
+    data = files['test/accounts.1.json']
+    assert open(fn, 'rb').read() == data
+    s3.put(fn, test_bucket_name+'/temp')
+    assert s3.du(test_bucket_name+'/temp')[test_bucket_name+'/temp'] == len(data)
+    assert s3.cat(test_bucket_name+'/temp') == data
```

Comment:
I use `shutil` in case we use the generated name to create a file.  `mktemp` has, I think, some threadsafety issues, mkstemp is preferred I think.  `os.remove(fn)` fails in Windows if the file is still open.

`dask.utils.py:tmpfile` started out as something similar to what you propose.  It grew as we ran into more and more complex testing situations which inevitably arise in the real world (open files, windows, directories, threads.)